zephyr/src/service.c

```diff
@@ -55,6 +55,38 @@ struct device *pwm, *tmp36;
 
 #define ADC_DEVICE_NAME "ADC_0"
 
+#define SLEEPTICKS      SECONDS(1)
+
+void pwmloop(void)
+{
+	if ((pwm = device_get_binding("PWM")) == NULL)
+        printk("device_get_binding: failed for PWM\n");
+	else
+        printk("PWM init OK\n");
+
+    while(true) {
+		task_sleep(SLEEPTICKS);
+        uint32_t r_on, g_on, b_on;
+        uint32_t r_off, g_off, b_off;
+
+        r_on = rgb[0] * 4;
+        g_on = rgb[1] * 4;
+        b_on = rgb[2] * 4;
+
+        r_off = 1024 - r_on;
```

Comment:
as far as I understand now, this probably works just fine with 1024, but works at a very high frequency. (on + off) time details the frequency, and 32000 is 1ms according to the examples.

1024 is thus 0.032ms, meaning 31250 Hz

Arduino apparently uses 500Hz for PWM (500 Hz = 2 ms/period)

500Hz is 2ms, thus 64000 in total for on+off. Thus we should probably use 64000 instead of 1024 here.